File: frontend/src/lib/schemas/deep_research.ts
import type { Optional } from '@sveltejs/kit';

export interface DeepResearchRequest {
    query: string;
    chat_id?: Optional<string>;
    max_concurrent_research_units?: Optional<number>;
    max_researcher_iterations?: Optional<number>;
    max_react_tool_calls?: Optional<number>;
    max_structured_output_retries?: Optional<number>;
}

export interface DeepResearchResponse {
    report: string;
}

File: backend/app/schemas/deep_research.py

from pydantic import BaseModel, Field
from typing import Optional
from uuid import UUID

class DeepResearchRequest(BaseModel):
    query: str
    chat_id: Optional[UUID] = None
    max_concurrent_research_units: Optional[int] = Field(default=1)
    max_researcher_iterations: Optional[int] = Field(default=1)
    max_react_tool_calls: Optional[int] = Field(default=1)
    max_structured_output_retries: Optional[int] = Field(default=1)

class DeepResearchResponse(BaseModel):
    report: str
